### PR TITLE
chore: Documentation and test code movement around topology

### PR DESCRIPTION
**Pull request by blt**
As a part of #8512 I need to have a much more clear understanding of the
topology construct of vector than I presently do. I have an eye toward
accomplishing some of the aims of the Architecture Revisit RFC but would like to
get there in small steps.

This commit does not functionally change vector. It moves some test code around
and adjusts the documentation to be more, I hope, clear about what's what.

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

File: src/topology/test/doesnt_reload.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    topology
+        .reload_config_and_respawn(new_config.build().unwrap())
+        .await
+        .unwrap();
+
+    assert_eq!(
+        topology.config.global.data_dir,
+        Some(Path::new("/asdf").to_path_buf())
+    );
```
Comment (tobz) [inline src/topology/test/doesnt_reload.rs:36]: I have to admit, this doesn't make sense to me. Why wouldn't the updated configuration be present?  Kind of a weird test at face value.

**Review by tobz — APPROVED** [review]
LGTM.  I left a stream-of-consciousness comment unrelated to the validity of this PR about one of the tests.

File: src/topology/test/doesnt_reload.rs (hunk 1 above)
Comment (blt) [reply src/topology/test/doesnt_reload.rs:36]: I don't know my own self. I do intend to take a pass over the tests in future work, so I've made a note of this question.

Comment (lukesteensen) [reply src/topology/test/doesnt_reload.rs:36]: We explicitly don't want to support changing the data directory at runtime. Original PR is [here](https://github.com/timberio/vector/pull/193) for (some) context.

**Pull request merged**